src/d3a/models/market/market_structures.py

```diff
@@ -177,17 +177,40 @@ def offer_or_bid_from_JSON_string(offer_or_bid, current_time):
         return Bid(**offer_bid_dict)
 
 
-class TradeBidInfo(namedtuple('TradeBidInfo',
-                              ('original_bid_rate', 'propagated_bid_rate',
-                               'original_offer_rate', 'propagated_offer_rate',
-                               'trade_rate'))):
+def _is_offer(offer_or_bid):
+    offer_bid_dict = json.loads(offer_or_bid)
+    object_type = offer_bid_dict.pop("type")
+    if object_type == "Offer":
+        return True
+    else:
+        return False
+
+
+def _is_bid(offer_or_bid):
```

Comment:
But also: I am not seeing `_is_bid` and `is_offer` being used anywhere. Is this a preparation for the future?

Reply:
Good point, there were used before refactoring, I'll clean that up, thanks